great_expectations/data_context/types/resource_identifiers.py

```diff
@@ -202,12 +202,16 @@ def from_object(cls, validation_result):
 
 class GeCloudIdentifier(DataContextKey):
     def __init__(
-        self, resource_type: "GeCloudRESTResource", ge_cloud_id: Optional[str] = None
+        self,
+        resource_type: "GeCloudRESTResource",
+        ge_cloud_id: Optional[str] = None,
+        resource_name: Optional[str] = None,
```

Comment:
Added resource_name so that we can use the name as an identifier in addition to the id